v2/emailpassword/nextjs/app-directory/protecting-route.mdx

```diff
@@ -177,28 +177,34 @@ import { getSSRSession } from "supertokens-node/nextjs";
 import { SessionContainer } from "supertokens-node/recipe/session";
 import { cookies, headers } from "next/headers";
 declare let SessionAuthForNextJS: any; // typecheck-only, removed from output
-import styles from "../page.module.css";
 import { redirect } from "next/navigation";
 // @ts-ignore
 import { TryRefreshComponent } from "./tryRefreshClientComponent";
 // @ts-ignore
 import { SessionAuthForNextJS } from "./sessionAuthForNextJS";
 
-async function getSession(): Promise<{ session: SessionContainer | undefined; hasToken: boolean; hasInvalidClaims: boolean }> {
+async function getSSRSessionHelper(): Promise<{ session: SessionContainer | undefined; hasToken: boolean; hasInvalidClaims: boolean, error: Error | undefined }> {
     let session: SessionContainer | undefined;
     let hasToken = false;
     let hasInvalidClaims = false;
+    let error: Error | undefined = undefined;
     
     try {
         ({ session, hasToken, hasInvalidClaims } = await getSSRSession(cookies().getAll(), headers()));
```

Comment:
```suggestion
        { session, hasToken, hasInvalidClaims } = await getSSRSession(cookies().getAll(), headers());
```

Reply:
It is required!

<img width="1353" alt="image" src="https://github.com/supertokens/docs/assets/22813027/28f119f0-bead-49c9-8f7f-ec55ca221343">